server/src/main/java/org/opensearch/tasks/BackgroundTaskService.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tasks;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.lifecycle.Lifecycle;
+import org.opensearch.common.lifecycle.LifecycleComponent;
+import org.opensearch.common.lifecycle.LifecycleListener;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.util.concurrent.AbstractAsyncTask;
+import org.opensearch.offline_tasks.TaskClient;
+import org.opensearch.offline_tasks.task.Task;
+import org.opensearch.offline_tasks.task.TaskType;
+import org.opensearch.offline_tasks.worker.TaskWorker;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Service to manage execution of tasks on Offline Node
+ */
+public class BackgroundTaskService implements LifecycleComponent {
+
+    private final Logger logger = LogManager.getLogger(BackgroundTaskService.class);
+    private final ThreadPool threadPool;
+    private AsyncBackgroundTaskFetch backgroundTaskFetch;
+    /**
+     * TaskClient used to interact with TaskStore/Queue for Task Management and Coordination
+     */
+    private final TaskClient taskClient;
+    /**
+     * Provides Corresponding TaskWorker to execute for each Task
+     */
+    private final Map<TaskType, TaskWorker> taskWorkerMap;
+
+    public BackgroundTaskService(ThreadPool threadPool, TaskClient taskClient, Map<TaskType, TaskWorker> taskWorkerMap) {
+        this.threadPool = threadPool;
+        this.taskClient = taskClient;
+        this.taskWorkerMap = taskWorkerMap;
+    }
+
+    @Override
+    public void start() {
+        backgroundTaskFetch = new AsyncBackgroundTaskFetch(this);
+    }
+
+    @Override
+    public void stop() {
+        backgroundTaskFetch.close();
+        backgroundTaskFetch.cancel();
+    }
+
+    @Override
+    public void close() {
+        stop();
+    }
+
+    @Override
+    public Lifecycle.State lifecycleState() {
+        return null;
+    }
+
+    @Override
+    public void addLifecycleListener(LifecycleListener listener) {
+
+    }
+
+    @Override
+    public void removeLifecycleListener(LifecycleListener listener) {
+
+    }
+
+    public void maybeExecuteTasks() {
+        List<Task> unassignedTasks = taskClient.getUnassignedTasks();
+        for (Task task : unassignedTasks) {
+            if (taskWorkerMap.containsKey(task.getTaskType())) {
+                if (taskClient.claimTask(task.getTaskId())) {
+                    taskWorkerMap.get(task.getTaskType()).executeTask(task);
+                }
+            }
+        }
+    }
+
+    private class AsyncBackgroundTaskFetch extends AbstractAsyncTask {
```

Comment:
Should we name this `BackgroundTaskFetch` or `AsyncTaskFetch`, unless the current name is intentional?